`profile.c` caused build regression

Build fails on Fedora with GCC-14.1.1 since 8d95f3c. Reverting to 495e60d builds fine.

Line 418 in profile.c expected a `const short unsigned int *` but got `unsigned char *` tripping `-Wincompatible-pointer-types` thus failing compilation.

```
cc -Wall -O3 -I src -std=c11 -D LIBHL_EXPORTS -m64 -fPIC -pthread -fno-omit-frame-pointer -o src/profile.o -c src/profile.c
src/profile.c: In function ‘profile_dump’:
src/profile.c:418:70: error: passing argument 1 of ‘hl_to_utf8’ from incompatible pointer type [-Wincompatible-pointer-types]
  418 |         char* filename = ptr == NULL ? "hlprofile.dump" : hl_to_utf8(ptr);
      |                                                                      ^~~
      |                                                                      |
      |                                                                      vbyte * {aka unsigned char *}
In file included from src/profile.c:22:
src/hl.h:604:39: note: expected ‘const uchar *’ {aka ‘const short unsigned int *’} but argument is of type ‘vbyte *’ {aka ‘unsigned char *’}
  604 | HL_API char *hl_to_utf8( const uchar *bytes );
      |                          ~~~~~~~~~~~~~^~~~~
make: *** [Makefile:291: src/profile.o] Error 1
```

--- a/src/profile.c
+++ b/src/profile.c
@@ -415,7 +415,7 @@ static void profile_dump( vbyte* ptr ) {
 	printf("Writing profiling data...\n");
 	fflush(stdout);
 
-	char* filename = ptr == NULL ? "hlprofile.dump" : hl_to_utf8(ptr);
+	char* filename = ptr == NULL ? "hlprofile.dump" : hl_to_utf8((uchar*)ptr);
 	FILE *f = fopen(filename,"wb");
 	int version = HL_VERSION;
 	fwrite("PROF",1,4,f);

